### PR TITLE
Command line arguments for running as server or client

### PR DESCRIPTION
**Pull request by NancokPS2**
Allows using "j_client" or "j_server" to instantly launch the game in either mode (it simply presses either of the buttons at startup by itself)  
  
I am currently unable to test it due to exports not working (or at least me not knowing how to make them work). I can confirm that the arguments work, but the game crashes shortly after due to being unable to load env variables (which should be unrelated).  
To test it, either run the game from CMD with the cmd arguments added after the path of the executable, or create a shortcut where at the end of the file path, specifies arguments separated by spaces  
  
Example of a working "Destination" on Windows:  
  
`C:\ProgramFiles\JDungeon\JDungeon.exe j_client`

### CODE REVIEW

**Comment by jonathaneeckhout** [comment]
This is great. We will need this to autodeploy the server on a server.

File: scenes/root/Root.gd [hunk 1]
```diff
@@ -71,6 +73,20 @@ func register_items():
...
+	var args:PackedStringArray = OS.get_cmdline_args()
+	if not args.is_empty():
+		J.logger.info("Found launch arguments. ", str(args))
+		
+	for arg in args:
+		if arg == "j_client":
+			$SelectRunMode/VBoxContainer/RunAsClientButton.pressed.emit()
+			break
+			
+		elif "j_server" in args:
```
Comment (jonathaneeckhout) [inline scenes/root/Root.gd:86]: Isn't this weird to do the "in" check of "args" while looping over "args"?

I think it's best to pick one mechanism.  

Option 1:
```
for arg in args:
    if arg == "j_client":
        start client
        return
    elif arg == "j_server":
        start server
        return
```

Option 2:
```
if "j_client" in args:
   start client
    return
elif "j_server" in args:
    start server
    return
```

Comment (NancokPS2) [reply scenes/root/Root.gd:86]: Yep, it is a remainder from an earlier failed approach.  
Will fix later today.

**Comment by NancokPS2** [comment]
Now it's more readable and scalable.

**Comment by jonathaneeckhout** [comment]
Tested it and it works. However it did expose some issues in the jsonbackend. I will merge and create a bug for the json backend.

**Pull request merged**